# HG changeset patch
# User msirringhaus@suse.de
# Date 1558442998 -7200
#      Tue May 21 14:49:58 2019 +0200
# Node ID b9d49f4c03a4801c3d88e3c10fde24bfb79231ff
# Parent  9a0fe18c7b744a15228ef1683d4bd549dc6e2380
bsc#991344 - Rpi3: Firefox crashes after a few seconds of usage
bmo#1302554 - ARM/AARCH64: Firefox crashes on NULL nsIChannel** result pointer in nsIOService::NewChannelFromURIWithProxyFlagsInternal()

diff -r 9a0fe18c7b74 -r b9d49f4c03a4 netwerk/base/nsIOService.cpp
--- a/netwerk/base/nsIOService.cpp	Fri Feb 26 16:20:09 2016 +0000
+++ b/netwerk/base/nsIOService.cpp	Tue May 21 14:49:58 2019 +0200
@@ -1000,7 +1000,13 @@
     }
   }
 
+#if defined(__aarch64__)
+  if (result) {
+    channel.forget(result);
+  }
+#else
   channel.forget(result);
+#endif
   return NS_OK;
 }
 
